File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    
    unordered_map<string, int> hashmap;
    
    for(const auto &name : participant){
        
        hashmap[name]++;
        
    }
    
    for(const auto &name: completion){
        
        hashmap[name]--;
        if (hashmap[name] == 0) hashmap.erase(name);
        
    }

    answer = hashmap.begin()->first;
    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    
    part_dict = dict()
    for part in participant:
        try:
            part_dict[part] +=1
        except:
            part_dict[part] =1 
    
    for comp in completion:
        part_dict[comp] -= 1
        

    for k, v in part_dict.items():
        if v == 1:
            return k